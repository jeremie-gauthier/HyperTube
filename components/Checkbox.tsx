import { FlexRow } from "@/components/Flex";
import styles from "./Checkbox.module.scss";

type CheckboxProps = {
  label: string;
  name: string;
} & Partial<React.InputHTMLAttributes<HTMLInputElement>>;

export default function Checkbox({
  label,
  ...rest
}: CheckboxProps): JSX.Element {
  return (
    <FlexRow className={styles.container}>
      <input type="checkbox" {...rest} />
      <label htmlFor={rest.name}>{label}</label>
    </FlexRow>
  );
}
