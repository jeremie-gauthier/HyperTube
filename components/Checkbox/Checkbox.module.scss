.container {
  @apply items-center space-x-2 select-none text-sm;

  input[type="checkbox"] {
    @apply h-4 w-4 appearance-none outline-none shadow-none text-center cursor-pointer rounded;

    line-height: 1em;

    &:focus {
      @apply bg-white;
    }
  }

  input[type="checkbox"]:checked::after {
    content: "âœ”";
    color: #000;
  }
}
