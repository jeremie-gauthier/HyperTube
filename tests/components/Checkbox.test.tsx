import React from "react";
import Checkbox from "@/components/Checkbox";
import renderer from "react-test-renderer";

test("<Checkbox /> should render correctly", () => {
  const tree = renderer.create(<Checkbox label="oui" name="yesno" />).toJSON();
  expect(tree).toMatchSnapshot();
});
