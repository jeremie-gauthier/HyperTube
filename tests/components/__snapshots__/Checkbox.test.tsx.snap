// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render correctly 1`] = `
<div
  className="flexRow container"
>
  <input
    name="yesno"
    type="checkbox"
  />
  <label
    htmlFor="yesno"
  >
    oui
  </label>
</div>
`;
